extensions/sql-bindings/src/common/azureFunctionsUtils.ts

```diff
@@ -594,3 +633,59 @@ export async function promptSelectDatabase(connectionInfo: IConnectionInfo): Pro
 	}
 	return selectedDatabase;
 }
+
+export async function getConnectionURI(connectionInfo: IConnectionInfo): Promise<string | undefined> {
+	const vscodeMssqlApi = await utils.getVscodeMssqlApi();
+
+	let connectionURI: string = '';
+	try {
+		await vscode.window.withProgress(
+			{
+				location: vscode.ProgressLocation.Notification,
+				title: constants.connectionProgressTitle,
+				cancellable: false
+			}, async (_progress, _token) => {
+				// show progress bar while connecting to the users selected connection profile
+				connectionURI = await vscodeMssqlApi.connect(connectionInfo!);
+			}
+		);
+	} catch (e) {
+		// mssql connection error will be shown to the user
+		return undefined;
+	}
+
+	return connectionURI;
+}
+
+export async function promptSelectTable(connectionURI: string, selectedDatabase: string, bindingType: BindingType): Promise<string | undefined> {
+	const vscodeMssqlApi = await utils.getVscodeMssqlApi();
+	const userObjectName = bindingType === BindingType.input ? constants.enterObjectName : constants.enterObjectNameToUpsert;
```

Comment:
It would be good to have an icon for this too to clearly distinguish it from the actual table names. The pencil icon seems like a good choice https://code.visualstudio.com/api/references/icons-in-labels

Reply:
<img width="423" alt="image" src="https://user-images.githubusercontent.com/23587151/168406469-bc1d72b1-a196-47e5-8df7-3c5acdace61c.png">